sdk/eventgrid/azure-messaging-eventgrid/src/main/java/com/azure/messaging/eventgrid/EventGridEvent.java

```diff
@@ -67,27 +57,15 @@ public EventGridEvent(String subject, String eventType, Object data, String data
     }
 
     /**
-     * Parse the EventGrid Event from a JSON string. This can be used to interpret the event at the event destination
-     * from raw JSON into rich event(s).
-     * @param json the JSON payload containing one or more events.
+     * Deserialize the {@link EventGridEvent} from a JSON string.
+     * @param eventGridJsonString the JSON payload containing one or more events.
      *
-     * @return all of the events in the payload parsed as CloudEvents.
+     * @return all of the events in the payload deserialized as {@link EventGridEvent}s.
+     * @throws IllegalArgumentException if the input parameter isn't a JSON string for a eventgrid event
+     * or an array of it.
      */
-    public static List<EventGridEvent> parse(String json) {
-        return Flux.fromArray(deserializer
-            .deserialize(new ByteArrayInputStream(json.getBytes(StandardCharsets.UTF_8)),
-                TypeReference.createInstance(com.azure.messaging.eventgrid.implementation.models.EventGridEvent[].class))
-            )
-            .map(event -> {
-                if (event.getData() == null) {
-                    return new EventGridEvent(event);
-                }
-                ByteArrayOutputStream stream = new ByteArrayOutputStream();
-                deserializer.serialize(stream, event.getData());
-                return new EventGridEvent(event.setData(stream.toByteArray())); // use BinaryData instead?
-            })
-            .collectList()
-            .block();
+    public static List<EventGridEvent> fromString(String eventGridJsonString) {
+        return EventGridDeserializer.deserializeEventGridEvents(eventGridJsonString);
```

Comment:
What happens when `eventGridJsonString` is null , should we check it ?

Reply:
Yes NPE will be thrown from the underlying library. So we may not need to check it. Added throws section in method doc